packages/react/menu/src/Menu.tsx

```diff
@@ -220,67 +227,76 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
-
-                      if (event.defaultPrevented) {
-                        // reset this because the event was prevented
-                        setIsPermittedPointerDownOutsideEvent(false);
-                      }
-                    },
-                    { checkForDefaultPrevented: false }
-                  )}
-                  onFocusOutside={composeEventHandlers(
-                    onFocusOutside,
-                    (event) => {
-                      // When focus is trapped, a focusout event may still happen.
-                      // We make sure we don't trigger our `onDismiss` in such case.
-                      if (trapFocus) event.preventDefault();
-                    },
-                    { checkForDefaultPrevented: false }
-                  )}
-                  onInteractOutside={onInteractOutside}
-                  onDismiss={() => context.onOpenChange(false)}
-                >
-                  {(dismissableLayerProps) => (
+            {(focusScopeProps) => (
+              <DismissableLayer
+                disableOutsidePointerEvents={disableOutsidePointerEvents}
+                onEscapeKeyDown={onEscapeKeyDown}
+                onPointerDownOutside={composeEventHandlers(
+                  onPointerDownOutside,
+                  (event) => {
+                    const isLeftClick =
+                      (event as MouseEvent).button === 0 && event.ctrlKey === false;
+                    const isPermitted = !disableOutsidePointerEvents && isLeftClick;
+                    setIsPermittedPointerDownOutsideEvent(isPermitted);
+
+                    if (event.defaultPrevented) {
+                      // reset this because the event was prevented
+                      setIsPermittedPointerDownOutsideEvent(false);
+                    }
+                  },
+                  { checkForDefaultPrevented: false }
+                )}
+                onFocusOutside={composeEventHandlers(
+                  onFocusOutside,
+                  (event) => {
+                    // When focus is trapped, a focusout event may still happen.
+                    // We make sure we don't trigger our `onDismiss` in such case.
+                    if (trapFocus) event.preventDefault();
+                  },
+                  { checkForDefaultPrevented: false }
+                )}
+                onInteractOutside={onInteractOutside}
+                onDismiss={() => context.onOpenChange(false)}
+              >
+                {(dismissableLayerProps) => (
+                  <RovingFocusGroup
+                    as={Slot}
+                    dir={dir}
+                    orientation="vertical"
+                    loop={loop}
+                    currentTabStopId={currentItemId}
+                    onCurrentTabStopIdChange={setCurrentItemId}
+                    // we override the default behaviour which automatically focuses the first item
+                    onEntryFocus={(event) => event.preventDefault()}
```

Comment:
We have that name in `FocusScope` yes, but that's because it's truly on mount.
This here, doesn't have anything to do with mount or unmount, it's the refocus that happens when the wrapper gets focused.

Reply:
So is it just `onFocusEnter` like `onMouseEnter` ? When focus enters the roving focus group we want to prevent it?

Reply:
Well that's the thing, `onFocusEnter` just sounds like `onFocus`, which is the event we hook onto. But because this is a custom event, I'd rather want to name it according to what it does, rather than what event it hooks on so.
If I describe what the event does (with its associated behaviour that you can prevent) is:

> when the wrapper itself gets focused, move focus to intended item

That's what the name should be based on.

Reply:
`onFocusEnter` isn't `onFocus` in my mind because React `onFocus` is actually `onFocusIn` 🙈  I'm imagining `onMouseEnter` which doesn't bubble, so it is only triggered when mouse first enters the element and not its children. I was applying that same thinking here, so `onFocusEnter` would be when focus first enters the wrapper and not when it's focusing children. Isn't that what this is?

Reply:
I get that, but my previous comment was more that I think the event should reflect the specific "what", rather than the "when", seeing as that's the behaviour we want to cancel by "preventDefaulting"ing it.

I know one could say the opposite for example, that on a `select`, arrow down to open isn't an event called `onArrowDownOpen` that you prevent, but you'd prevent the keydown when the key is arrow (ie. it's more about the "when" than the "why") but then if we want to go for that, then I'm not sure why we need a custom event at all, we would just stay on what I had originally which was to `preventDefault` in `onFocus`.

What do you think?

Reply:
Interesting. Didn't we move away from `onFocus` because that would mean preventing browser behaviour too? I'm good to stick with `onEntryFocus` :+1:

Reply:
Well, there isn't anything to prevent natively `onFocus` but the reason we moved away was rather that it seemed like implicit knowledge to have the `Menu` component prevent default in `onFocus` just because we happen to know that the `RovingFocusGroup` is doing something in the `onFocus`.

But yeah I think it's enough bikeshedding here considering that it's not really public API anyway.